openslides/utils/collection.py

```diff
@@ -117,14 +152,217 @@ def get_full_data(self):
         Returns the full_data of this collection_element from with all other
         dics can be generated.
         """
-        return self.get_access_permissions().get_full_data(self.get_instance())
+        # If the full_data is already loaded, return it
+        # If there is a db_instance, use it to get the full_data
+        # else: try to use the cache.
+        # If there is no value in the cach, get the content from the db and save
+        # it to the cache.
+        if self.full_data is None and self.instance is None:
+            # Use the cache version if self.instance is not set.
+            # After this line full_data can be None, if the element is not in the cache.
+            self.full_data = cache.get(self.get_cache_key())
+
+        if self.full_data is None:
+            self.full_data = self.get_access_permissions().get_full_data(self.get_instance())
+            self.save_to_cache()
+        return self.full_data
 
     def is_deleted(self):
         """
         Returns Ture if the item is marked as deleted.
         """
         return self.deleted
 
+    def get_cache_key(self):
```

Comment:
Proposal: Move caching methods (and especially cache key and cache key prefix functions) to an extra class `Cache` for better code structure and readability.